core/src/main/java/org/elasticsearch/search/sort/ScriptSortBuilder.java

```diff
@@ -216,6 +217,32 @@ public XContentBuilder toXContent(XContentBuilder builder, Params builderParams)
         return builder;
     }
 
+    private static ConstructingObjectParser<ScriptSortBuilder, QueryParseContext> PARSER = new ConstructingObjectParser<>(NAME,
+            a -> new ScriptSortBuilder((Script) a[0], (ScriptSortType) a[1]));
+
+    static {
+        PARSER.declareField(constructorArg(), (p, c) -> {
+            try {
+                return Script.parse(p, c.getParseFieldMatcher(), c.getDefaultScriptLanguage());
```

Comment:
It'd be lovely if `script` had static  `parser` method that returned the right `Function` for use here.